torch/_functorch/partitioners.py

```diff
@@ -837,15 +840,27 @@ def get_node_weight(node) -> int:
         # Heuristic to bias towards nodes closer to the backwards pass
         # Complete guess about current value
         mem_sz = int(mem_sz * (1.1 ** max(min(node.dist_from_bw, 100), 1)))
-        # mem_sz = int(mem_sz + node.dist_from_bw)
-
         if is_materialized(node):
             return mem_sz
         else:
             return mem_sz * 2
 
     nx_graph = nx.DiGraph()
-    for node in full_bw_graph.nodes:
+    banned_nodes = set()
+
+    def ban_recomputation_if_allowed(node):
+        # NB: "recompute" == 0 means it's not banned from recomputing
+        if node.meta.get("recompute", 0) > 0:
+            return False
+        banned_nodes.add(node)
+        # A node will only ever be recomputed if there is a path from an
+        # ancestor of this node to the backwards path through this node that
+        # doesn't go through any saved value. If this node is saved, then that
+        # condition is not possible.
+        nx_graph.add_edge("source", node.name + "_in", capacity=math.inf)
```

Comment:
worth factoring out the inf banning of recomputation into helper ?

Reply:
This is the helper haha.